### PR TITLE
feat(ui): add page level loading layer

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add global page level loading layer

add page level loading layer

## Linear Issue Reference

LOG-2091

## Testing

test locally
@logto-io/eng

### CODE REVIEW

File: packages/ui/src/components/Icons/LoadingIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { SVGProps } from 'react';
+
+import Loading from '@/assets/icons/loading.svg';
+
+const LoadingIcon = (props: SVGProps<SVGSVGElement>) => {
+  return (
+    <svg width="32" height="32" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg" {...props}>
+      <use href={`${Loading}#loading`} />
```
Comment (demonzoo) [inline packages/ui/src/components/Icons/LoadingIcon.tsx:8]: What's good of nesting the loading.svg inside, comparing to directly writing the whole svg content here?

File: packages/ui/src/components/LoadingLayer/index.module.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+.container {
+  width: 60px;
+  height: 60px;
+  border-radius: 8px;
+  background-color: var(--color-dark-background);
+  @include _.flex-column;
+}
+
+.loadingIcon {
+  animation: rotating 1.5s ease-in-out infinite;
```
Comment (demonzoo) [inline packages/ui/src/components/LoadingLayer/index.module.scss:22]: use `linear`?

**Review by wangsijie — APPROVED** [review]

File: packages/ui/src/hooks/use-api.ts [hunk 3]
```diff
@@ -16,6 +18,12 @@ function useApi<Args extends any[], Response>(
   const [error, setError] = useState<RequestErrorBody>();
   const [result, setResult] = useState<Response>();
 
+  const { setLoading: setPageLoading } = useContext(PageContext);
+
+  useEffect(() => {
+    setPageLoading(loading);
+  }, [loading, setPageLoading]);
```
Comment (demonzoo) [inline packages/ui/src/hooks/use-api.ts:25]: This looks weird. If we have setPageLoading in the page context, why would we declare another loading state in the hook?

File: packages/ui/src/components/Icons/LoadingIcon.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/components/Icons/LoadingIcon.tsx:8]: Yeah just to keep the original SVG files together under the assets path.  I am expecting eventually no need to provide these hand written svg components.

File: packages/ui/src/hooks/use-api.ts (hunk 3 above)
Comment (simeng-li) [reply packages/ui/src/hooks/use-api.ts:25]: good call I was thinking to keep each API's original loading status and a global loading status. But there are issues you mentioned before. Each API will overwrite the global status async. I'll simply remove it.

File: packages/ui/src/components/Icons/LoadingIcon.tsx (hunk 1 above)
Comment (demonzoo) [reply packages/ui/src/components/Icons/LoadingIcon.tsx:8]: OK, I see your points. Interesting... Maybe we should do this in admin console as well

File: packages/ui/src/components/LoadingLayer/index.module.scss (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/components/LoadingLayer/index.module.scss:22]: Keep it for now, I noticed most of the loading spin effects has a pausing effect more like ease-in-out. Anyway will check the final animation solution with Eric.

Comment (demonzoo) [reply packages/ui/src/components/LoadingLayer/index.module.scss:22]: Yeah, I see. np

File: packages/ui/src/hooks/use-api.ts (hunk 3 above)
Comment (demonzoo) [reply packages/ui/src/hooks/use-api.ts:25]: np

Comment (simeng-li) [reply packages/ui/src/hooks/use-api.ts]: Updated

File: packages/ui/src/components/Icons/LoadingIcon.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/components/Icons/LoadingIcon.tsx:8]: k, leave it just for now

**Pull request merged**